os/fs/smartfs/smartfs_smart.c

```diff
@@ -1301,6 +936,7 @@ static int smartfs_truncate(FAR struct file *filep, off_t length)
 	FAR struct smartfs_ofile_s *sf;
 	off_t oldsize;
 	int ret;
+	uint16_t oldfilepos;
```

Comment:
Are you referring to the warning saying "oldfilepos may be used uninitialized"?

Reply:
yes. some of tools which can detect defect will detect line 960 as a minor issue, it should be moved to below of branch state in line 962.

Reply:
Because errout_with_semaphore causes oldfiilepos to remain unitialized,
I moved oldfilepos = sf->savefilepos before Line 964.

So now if 964 goes to errout_with_semaphore, olldfilepos is initialized.

Reply:
Does that resolve your concern?